### ISSUE TITLE
Simplified marker generation

### ISSUE THREAD

**Issue by gaddman** [openchargemap/ocm-system#90]
Issue to work towards this comment in issue #89:
>Long term I'd like icons to be composited from a single marker shape, a color base applied and then any annotations applied on top (no entry, planned/under contruction, photos or checkins available etc). Ideally we'd do this from a single script rather than fiddling with the icons individually.

**Comment by gaddman** [comment]
So my git skills are not too hot. I've made a commit and then pull request for issue #89, then tried to do a separate pull request for this issue, on the same branch. The pull requests have merged into one. Hopefully you can choose which commits in that pull request you want to accept.

**Comment by webprofusion-chrisc** [comment]
Nice, very much looking forward to trying this script out. Wondering if I can get bash working through cygwin (I'm on windows). Or could the shell script be replaced by a python script? Thinking cross platform (not like me, I know).

**Comment by webprofusion-chrisc** [comment]
Tried it out, looks cool. Ideally I'd like to have a way to define a single main marker shape but apply the level colour dynamically (clone marker shape or a named colour layer maybe?) bit tricky to script though. That way the there would only be one copy of the marker design.

**Comment by gaddman** [comment]
Yeah, I thought this'll do as a starter, but agree there's got to be a better way. A couple of possibilities I can think of for now:
1. Implement a script extension to Inkscape to set colour and export. I presume this can be done, although it becomes less portable (application not OS) through reliance on Inkscape.
2. Set colour in the SVG file via script outside of Inkscape, similar to how the script above works for visibility. But I'd prefer to set/view the colours in Inkscape, so need to figure out a way to do that.

I'll have a look around, but let me know if you've got any preference on above approaches.

**Comment by webprofusion-chrisc** [comment]
Option 2 sounds like the right kind of thing. We could maybe name color swatches in the master svg file to relate to the power band/level (currently this is level but long term I think this will be some sort of kw scale) then pickup those colors for the marker output? Definitely tricky.
